### ISSUE TITLE
Volume is not updated in remote if changed in Music

### ISSUE THREAD

**Issue by tcarroll2** [hhas/Swift-AppleScriptObjC#2]
When adjusting the volume within the Music.app while a track is playing, the volume on the remote is not tracking the changes.

How would you make an observer to detect changes from within the app?

**Comment by hhas** [comment]
You could check what Distributed Notifications Music.app emits, e.g. I know it sends notifications of track changes, and if there’s one for volume changes then register a listener with `DistributedNotificationCenter` for that.

Chances are it doesn’t (DNs are cool but massively undersupported and underused), in which case all you can do is actively poll. Run a repeating `Timer` in your app that periodically (e.g. every 200ms) sends Music.app a `get volume` Apple event and updates your GUI with the result of that.

**Comment by tcarroll2** [comment]
Roger that.  I will have to look to see, but similar to your thoughts it is probably a longshot.  I will certainly do the timer thing to poll the volume if a DN doesn't exist.  Thank you for putting this out there!

**Comment by tcarroll2** [comment]
My intention with this was to implement a background music option for when the main app is not playing music, so I dug around and tried implementing a couple items like get current playlist and set playlist, but I couldn't get this to work.  I was getting a lot of errors, so I decided AppleScript is for AppleScripters and Swift is for Swifters and I just implemented a background player in my app that I can set a playlist and play/pause as I needed when my app is not playing the main music.  It also helps to not to have to rely on sandboxing, hardened runtime, and other permissions, or opening Music in the background.  It's all contained in my app.

Again, thanks for sharing this and opening my mind to other possibilities and options.
